plugins/secretstores/docker/docker.go

```diff
@@ -0,0 +1,83 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package docker
+
+import (
+	_ "embed"
+	"errors"
+	"os"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// mount directory created by Docker when using Docker Secrets
+const dockerSecretsDir string = "/run/secrets"
+
+type Docker struct {
+	ID string `toml:"id"`
+}
+
+func (*Docker) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initializes all internals of the secret-store
+func (d *Docker) Init() error {
+	if d.ID == "" {
+		return errors.New("id missing")
+	}
+	if _, err := os.Stat(dockerSecretsDir); os.IsNotExist(err) {
+		return errors.New("/run/secrets directory does not exist")
+	}
+	return nil
+}
+
+func (d *Docker) Get(key string) ([]byte, error) {
+	secretFile := dockerSecretsDir + "/" + key
+	value, err := os.ReadFile(secretFile)
+	if err != nil {
+		return nil, errors.New("cannot find the secret file under /run/secrets")
```

Comment:
Simply return the error here as the error could also be wrong permissions etc...
```suggestion
		return nil, err
```

Reply:
If we return the `err` what would the test case capture look like? at the moment, I have refactored the error message and I capture a case for a non-existent secrets file and compare it with the error message.

Reply:
To provide better test coverage:

```go
require.ErrorContains(t, err, "cannot read the secret's value under the directory", err)
```
is used. This will check for the main substring of the defined error using `fmt.Errorf` here